exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/ApiController.java

```diff
@@ -408,23 +435,20 @@ private static String hexEncodeTransactionMessage(TransactionMessage transaction
     @VisibleForTesting
     static final String BLOCKCHAIN_HEIGHT_PATH = BLOCKCHAIN_ROOT + "/height";
     @VisibleForTesting
-    static final String BLOCKCHAIN_BLOCK_HASHES_PATH = BLOCKCHAIN_ROOT + "/block";
+    static final String BLOCKCHAIN_BLOCK_HASHES_PATH = BLOCKCHAIN_ROOT + "/blocks";
     @VisibleForTesting
     static final String BLOCKCHAIN_BLOCKS_PATH = BLOCKCHAIN_ROOT + "/hashesToBlocks";
     @VisibleForTesting
     static final String BLOCK_HEIGHT_PARAM = "blockHeight";
     @VisibleForTesting
-    static final String BLOCKCHAIN_BLOCK_PATH = BLOCKCHAIN_ROOT + "/block/:" + BLOCK_HEIGHT_PARAM;
-    @VisibleForTesting
-    static final String BLOCKCHAIN_LAST_BLOCK_PATH = BLOCKCHAIN_ROOT + "/lastBlock";
+    static final String BLOCK_ID_PARAM = "blockId";
     @VisibleForTesting
-    static final String BLOCKCHAIN_BLOCK_TRANSACTIONS_BY_HEIGHT_PATH = BLOCKCHAIN_ROOT + "/block/:"
-        + BLOCK_HEIGHT_PARAM + "/transactionsByHeight";
+    static final String BLOCKCHAIN_BLOCK_PATH = BLOCKCHAIN_ROOT + "/block";
     @VisibleForTesting
-    static final String BLOCK_ID_PARAM = "blockId";
+    static final String BLOCKCHAIN_LAST_BLOCK_PATH = BLOCKCHAIN_ROOT + "/block/last";
```

Comment:
How about to replace `/last` by request param as well? `block?last=true` `block?height=max`
Also, with such filtering, we'll be able to fetch transactions for the last block by one call.

Reply:
I don't like the current one, but both options that you suggested don't look that good to me either :)
@dmitry-timofeev wdyt?

Reply:
😄

Reply:
Probably the most buetiful paths will be like below. But them requires to have a quite complicated logic of matchers for `heigth`, for `hash` and special word "last".

`GET /blockchain/blocks` -> returns all block hashes
`GET /blockchain/blocks/{height | hash | last}`  -> returns the block by height or hash or last block
`GET /blockchain/blocks/{height | hash | last}/transactions`   -> returns the block txs by height or hash or last block

Reply:
I'd stick with a simpler in terms of maintenance version, since this service has a single client.